packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -737,6 +739,35 @@ abstract class ResidentRunner {
     throw Exception('Canvaskit not supported by this runner.');
   }
 
+  /// Write the SkSL shaders to a zip file in build directory.
+  Future<void> writeSkSL() async {
+    if (!supportsWriteSkSL) {
+      throw Exception('writeSkSL is not supported by this runner.');
+    }
+    final Map<String, Object> data = await flutterDevices.first.views.first.getSkSLs();
+    if (data.isEmpty) {
+      globals.logger.printStatus(
+        'No data was receieved. To ensure SkSL data can be generated use a '
+        'physical device then:\n'
+        '  1. Pass "--cache-sksl" as an argument to flutter run.\n'
+        '  2. Interact with the application to force shaders to be compiled.\n'
+      );
+      return;
+    }
+    final File outputFile = globals.fsUtils.getUniqueFile(
```

Comment:
Just to confirm: if I run this 10 times with 10 devices, I'll get 10 different output files, each of which will be a plain text file with all the Base64 encoded SkSLs of that device, right?

Reply:
Another question: if I run this twice on the same device, does it generate two files with the exact same content? (If so, that looks a little duplicative.)

Reply:
Yes, both are correct. Are you concerned about management of these files? I am too, I'm not quite sure what a reasonable workflow for these files will be

Reply:
What if we group the SkSLs with the device id, and use the newer SkSL to rewrite the older SkSL if both the SkSL key and device id matches?

Reply:
Tracking and managing these files is a bit too high level for the command line API. once they are written a user can (and should) be allowed to move and rename them

Reply:
I'm curious what would happen for the next PR that package the SkSLs into assets. If each run generates a different json map of SkSLs, how are you going to merge those maps and put them into asset? Do you have to deduplicate by then since you can only have one SkSL file per key/filename?

Reply:
I think to start with, it would be fair to say that the tool can support including a single one of these files in the build, something like: 

`flutter build apk --release --include=sksl=path/tp/flutter2.sksl`

That allows for different shaders for iOS and Android, but requires them to be the same across Android versions.

If we wanted a workflow beyond that, I think it definitely needs more design work. I think that will be easier to do once the entire thing is wired up end to end.

Reply:
For example, I don't know of a way to do assets per device in an appbundle, only per ABI

Reply:
Sounds good. LGTM!